[Strapi 5] Breaking change for the removed webpack aliases (#2047)
strapi/documentation: docusaurus/docs/dev-docs/migration/v4-to-v5/breaking-changes/webpack-aliases-removed.md

---
title: Webpack Aliases are removed
description: A simplified approach of aliasing in Strapi v5. 
sidebar_label: Webpack Aliases removed
displayed_sidebar: devDocsMigrationV5Sidebar
tags:
 - breaking changes

>  - dependencies

Reviewer: I would add `webpack` as a tag, too :)

Author: That's true I forgot, thanks

---

import Intro from '/docs/snippets/breaking-change-page-intro.md'
import MigrationIntro from '/docs/snippets/breaking-change-page-migration-intro.md'

# Webpack Aliases are removed

In Strapi v5, webpack aliases are removed ensuring better compatibility and reduced dependency conflicts. 

<Intro />

## Breaking change description

<SideBySideContainer>

<SideBySideColumn>

**In Strapi v4**

- We maintained a specific list of dependencies that were aliased in webpack configuration. This ensured that plugins consistently used our versions of certain libraries like the design-system. <br/> <br/> However, this approach led to many bugs.

</SideBySideColumn>

<SideBySideColumn>

**In Strapi 5**

- In Strapi v5, we have simplified the aliasing process. Now, we only alias essential dependencies like `react`, `react-dom`, `react-router-dom`, and `styled-components`.

</SideBySideColumn>

</SideBySideContainer>

## Migration

<MigrationIntro />

### Notes

> - If you encounter issues with third-party plugins, we recommend opening an issue on the respective plugin's repository. Encourage them to add their dependencies to their `package.json` file to resolve compatibility issues.

Reviewer: ```suggestion
- If you encounter issues with third-party plugins, we recommend opening an issue on the respective plugin's repository. Encourage the plugin maintainers to add their dependencies to their `package.json` file to resolve compatibility issues.
```

- If you encounter issues with local plugins, you can fix them by amending the `resolve` option in your chosen bundler.

> ### Manual procedure

Reviewer: ```suggestion
### Manual procedure

To migrate to Strapi 5:
```

---
title: Webpack Aliases are removed
description: A simplified approach of aliasing in Strapi v5. 
sidebar_label: Webpack Aliases removed
displayed_sidebar: devDocsMigrationV5Sidebar
tags:
 - breaking changes
 - dependencies
 - webpack
---

import Intro from '/docs/snippets/breaking-change-page-intro.md'
import MigrationIntro from '/docs/snippets/breaking-change-page-migration-intro.md'

# Webpack Aliases are removed

In Strapi v5, webpack aliases are removed ensuring better compatibility and reduced dependency conflicts. 

<Intro />

## Breaking change description

<SideBySideContainer>

<SideBySideColumn>

**In Strapi v4**

> - We maintained a specific list of dependencies that were aliased in webpack configuration. This ensured that plugins consistently used our versions of certain libraries like the design-system. <br/> <br/> However, this approach led to many bugs.

Reviewer: ```suggestion
- We maintained a specific list of dependencies that were aliased in webpack configuration. This ensured that plugins consistently used our versions of certain libraries like the design-system.
```

I would remove this. No need to shoot ourselves in the foot 😂

Author: Okay cool
